src/Collector.cs

```diff
@@ -79,25 +83,36 @@ private void StartBackgroundWorker()
 
     private void WorkerOnDoWork(object sender, DoWorkEventArgs doWorkEventArgs)
     {
+      var failures = 0;
+      Socket socket = null;
+
+      // This is not really the worker loop, it is just very careful connect/authenticate
+      // If you want the worker loop, look at SendQueuedMessages
       while (!_worker.CancellationPending)
         {
-          Socket socket = null;
-          var failures = 0;
-
           try
             {
               socket = Connect();
-              Authenticate(socket);
-              failures = 0;
+              if(!Authenticate(socket))
+                {
+                  socket.Close();
+                  // Throttle calls to authenticate so that you do not flood the buffer if collector is down/slow for a few seconds.
+                  // Ultimately should share logic with the backoff stuff below, but did not want to just throw here.
+                  Thread.Sleep(5000);
```

Comment:
Sometimes it seems totally random to me which parts of the original style we keep and which parts we do not.  You have given me an excuse to do the thing that I resisted doing though, so I'll do that.